nevergrad/optimization/test_callbacks.py

```diff
@@ -112,3 +114,21 @@ def test_early_stopping() -> None:
     # below functions are inlcuded in the docstring
     assert optimizer.current_bests["minimum"].mean < 12
     assert optimizer.recommend().loss < 12  # type: ignore
+
+
+def test_optimization_logger(tmp_path: Path) -> None:
+    instrum = ng.p.Instrumentation(
+        None, 2.0, blublu="blublu", array=ng.p.Array(shape=(3, 2)), multiobjective=True
+    )
+    logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
+    logger = logging.getLogger(__name__)
+    optimizer = optimizerlib.OnePlusOne(parametrization=instrum, budget=100)
+    early_stopping = ng.callbacks.EarlyStopping(lambda opt: opt.num_ask > 3)
+    optimizer.register_callback(
+        "tell",
+        callbacks.OptimizationLogger(
+            logger=logger, log_level=logging.INFO, log_interval_tells=10, log_interval_seconds=0.1
+        ),
+    )
+    optimizer.register_callback("ask", early_stopping)
+    optimizer.minimize(_func, verbosity=2)
```

Comment:
can we capture the logs here and check that it actually works?

Reply:
Now checks that beginning of expected log message for 1st, 2nd and 3rd logs are correct. (doesn't check for correct results because they change every time.)

Reply:
you could check the full log if you seed the parametrization. But if you are confident that all the useful information is properly written then it's fine without.